app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt

```diff
@@ -210,6 +212,21 @@ class AwesomeBarView(
         return searchEngine
     }
 
+    /**
+     * Get a suggestions header if [currentEngineName] is the one of the default search engine.
+     *
+     * @param currentEngineName Name of the currently selected search engine.
+     */
+    private fun getSearchEngineSuggestionsHeader(currentEngineName: String?): String? {
+        val searchState = activity.components.core.store.state.search
+        val defaultSearchEngineName = searchState.selectedOrDefaultSearchEngine?.name
+
+        return when (defaultSearchEngineName == currentEngineName) {
```

Comment:
⚠️ This check should also check if `defaultSearchEngine != null`

Reply:
The `defaultSearchEngine != null` is handled in the `getSearchEngineSuggestionsHeader` from below
https://github.com/mozilla-mobile/fenix/blob/454e8fa572b0a4926baa872a05706d5b0d62c618/app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt#L224-L225

If `defaultSearchEngineName` is null then `getSearchEngineSuggestionsHeader` would also return null as we expect [based on this check](https://github.com/mozilla-mobile/fenix/blob/454e8fa572b0a4926baa872a05706d5b0d62c618/app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt#L198).

I can add a test for that.